Add org.virtualbox.VirtualBox

tulilirockz opened the pull request:
<!-- ⚠️⚠️  Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Please go to the preview tab to view the markdown below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

<!-- 💡 Please tick and write 'N/A' with a reason if a checklist item below is not applicable 💡 -->

- [X] Please describe the application briefly. VirtualBox is a general purpose virtualization software
- [X] Please attach a video showcasing the application on Linux using the Flatpak."
  - https://www.youtube.com/watch?v=UMQ5o-9vxGc
  - https://ursal.zone/@tulilirockz/115183340082130336
- [X] The Flatpak ID follows all the rules listed in the [Application ID requirements][appid].
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.
- [ ] I am an _(please keep whichever is applicable and remove the rest)_ author/developer/upstream contributor to the project.
      If not, I contacted upstream developers about this submission. I have not, this is why this is a Draft

<!-- 💡 Please mention below the GitHub usernames of any additional maintainers needed (if any) 💡 -->

<!-- ⚠️⚠️  Please DO NOT modify anything below this line ⚠️⚠️  -->

[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

tulilirockz committed:
com.oracle.Virtualbox: add package

Signed-off-by: Tulip Blossom <tulilirockz@proton.me>

tulilirockz committed:
virtualbox: cleanup manifest and install directly on manifest

Signed-off-by: Tulip Blossom <tulilirockz@proton.me>

Commit:
chore: metadata and script for releases

Signed-off-by: Tulip Blossom <tulilirockz@posteo.com>

tulilirockz commented:
bot, build

Commit:
chore: disable aarch64 for now

Signed-off-by: Tulip Blossom <tulilirockz@posteo.com>

hfiguiere commented on flathub.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "skip-arches": ["aarch64", "arm"]
```
```suggestion
  "only-arches": ["x86_64"]
```

Commit:
flathub: only x86 for now

Signed-off-by: Tulip Blossom <tulilirockz@posteo.com>

Commit:
virtualbox: use GLU from shared modules and fix permission error on lvm2 build

Signed-off-by: Tulip Blossom <tulilirockz@posteo.com>

tulilirockz commented:
bot, build

Commit:
fix(virtualbox): strip errors

Signed-off-by: Tulip Blossom <tulilirockz@posteo.com>

Commit:
fix(virtualbox): metainfo not getting installed + formatting errors

Signed-off-by: Tulip Blossom <tulilirockz@posteo.com>

tulilirockz commented:
bot, build

Commit:
fix(virtualbox): correctly install metainfo

Signed-off-by: Tulip Blossom <tulilirockz@posteo.com>

tulilirockz commented:
bot, build

Commit:
fix(virtualbox): wrong directory for icons

Signed-off-by: Tulip Blossom <tulilirockz@posteo.com>

fluoriteByte commented:
honestly, since i haven't actually opened virtualbox again since i tested ubuntu in it, i just noticed it literally did not save anything to the "VirtualBox VMs" folder, and seems to actively fail to access it even if its present;
tried it again while fiddling with flatseal, and adding persistent=`VirtualBox VMs` fixes it not saving the folder, and now it works fine

Matthias-adR commented:
bot, build

peerchemist commented:
I am testing this out, and I observe some visual glitches through the UI.

<img width="695" height="258" alt="image" src="https://github.com/user-attachments/assets/50bc5c97-297c-4fc9-9adc-86b4979110f3" />

<img width="687" height="280" alt="image" src="https://github.com/user-attachments/assets/16579716-c2b6-4bf5-9c98-1cdccd2a1d9e" />

In the console I see these error messages:

```
Gtk-Message: 17:28:35.716: Failed to load module "pk-gtk-module"
Gtk-Message: 17:28:35.716: Failed to load module "pk-gtk-module"
Qt WARNING: QObject::disconnect: wildcard call disconnects from destroyed signal of UIInvisibleWindow::unnamed
Qt WARNING: QObject::disconnect: wildcard call disconnects from destroyed signal of UIInvisibleWindow::unnamed
Qt WARNING: Cannot create accessible child interface for object:  UISelectorTreeView(0x564a9d069af0)  index:  6
Qt WARNING: Cannot create accessible child interface for object:  UISelectorTreeView(0x564a9d069af0)  index:  6
Qt WARNING: Cannot create accessible child interface for object:  UISelectorTreeView(0x564a9d069af0)  index:  6
Qt WARNING: Cannot create accessible child interface for object:  UISelectorTreeView(0x564a9d069af0)  index:  6
Qt WARNING: Cannot create accessible child interface for object:  UISelectorTreeView(0x564a9d069af0)  index:  6
Qt WARNING: Cannot create accessible child interface for object:  UISelectorTreeView(0x564a9d069af0)  index:  4
Qt WARNING: Cannot create accessible child interface for object:  UISelectorTreeView(0x564a9d069af0)  index:  4
Qt WARNING: Cannot create accessible child interface for object:  UISelectorTreeView(0x564a9d069af0)  index:  4
```

fluoriteByte committed:
Update to latest VirtualBox

fluoriteByte committed:
Ensure that it actually creates the VirtualBox folder before app startup, idk why this isn't
default flatpak behavior x.x

albertescanes commented:
> "com.oracle"? The official home of VirtualBox is [virtualbox.org](https://www.virtualbox.org/), so shouldn't it be org.virtualbox.virtualbox then?
> 
> Also, is there a benefit or a requirement I overlooked for the CamelCase package name? Makes typing the command harder.

org.virtualbox indeed seems like a more appropriate identifier, as the repository already contains some references using it. Ideally, even if the application doesn’t get verified, the AppStream files should be maintained by upstream.

tulilirockz committed:
Merge pull request #2 from fluoriteByte/com.oracle.VirtualBox

Update to latest VirtualBox + "--filesystem=VirtualBox VMs:create" thingy

hfiguiere commented on com.oracle.VirtualBox.metainfo.xml:
```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>com.oracle.VirtualBox</id>
```
rename this to org.virtualbox.VirtualBox

KAMiKAZOW replied:
Why CamelCase, though?

fluoriteByte replied:
oh alright i see :3

hfiguiere replied:
this only asked here is to use `org.virtualbox`. There rest is just noise.

albertescanes commented on com.oracle.VirtualBox.yaml:
```diff
@@ -0,0 +1,207 @@
+rename-desktop-file: virtualbox.desktop
+finish-args:
+  # Default path that VirtualBox places files at
+  - "--filesystem=~/VirtualBox VMs:create"
```
Wouldn't it be better to use --persist to avoid cluttering the home directories?

fluoriteByte replied:
id say its to mimic the normal behavior of virtualbox and make migrations easier, if a user wants to declutter they only need to add the flag

tulilirockz replied:
Thank you @fluoriteByte ! https://github.com/flathub/flathub/pull/6985/commits/ea2e6ed24b5903d93611ff09f979e653efca9f51

fluoriteByte committed:
the great renaming

fluoriteByte committed:
change --filesystem to --persist

tulilirockz committed:
Merge pull request #3 from fluoriteByte/thegreatrenaming

the great renaming + --persist thing

tulilirockz commented:
bot, build

hfiguiere commented on org.virtualbox.VirtualBox.yaml:
```diff
@@ -0,0 +1,207 @@
+rename-desktop-file: virtualbox.desktop
+finish-args:
+  # Default path that VirtualBox places files at
+  - "--persist=~/VirtualBox VMs"
```
```suggestion
  - "--persist=VirtualBox VMs"
```

it's always relative to $HOME

fluoriteByte replied:
oh yeah oops i did this change without realising i left the ~ x.x

tulilirockz replied:
https://github.com/flathub/flathub/pull/6985/commits/8b4b001dbdd0eecdc2217ee643a720df1a0524bd

hfiguiere commented on org.virtualbox.VirtualBox.yaml:
```diff
@@ -0,0 +1,207 @@
+app-id: org.virtualbox.VirtualBox
+runtime: org.kde.Platform
+runtime-version: "6.9"
```
current is 6.10

tulilirockz commented on org.virtualbox.VirtualBox.yaml:
```diff
@@ -0,0 +1,207 @@
+    buildsystem: simple
+    build-options:
+      env:
+        LIBRARY_PATH: /app/lib/X11:/app/lib
```
I'll gather logs to show you, it just does not compile without this for some really weird reason

tulilirockz committed:
fix(org.virtualbox.Virtualbox): update to 6.10 runtime

- YASM requires a patch for gcc-15
- updated lvm2
- comment around build system quirk
- fix `--persist` from review

Signed-off-by: Tulip Blossom <tulilirockz@outlook.com>

tulilirockz commented on org.virtualbox.VirtualBox.yaml:
```diff
@@ -0,0 +1,207 @@
+        build-options:
+          make-args:
+            # Necessary because dependency resolution is not declared properly upstream
+            - -j1
```
https://github.com/flathub/flathub/pull/6985/commits/8b4b001dbdd0eecdc2217ee643a720df1a0524bd

bbhtt replied:
This was not changed to no-parallel-make.

fluoriteByte commented:
i think it if we are going to do it in an unofficial capacity we might also want to rebrand it, cus i am still concerned by virtualbox's trademark terms lol

sideeffffect commented:
@AlexanderEichner @bird2tori @alflr @Dsen-orcl @ramshven @klaus-vb @galitsyn-oracle @thatnerdjack @alexander-rudnev @baserkan @VP-at-oracle
Hello guys, thanks a lot for working on Virtualbox. Just so that you know, we're adding it to flathub.

bbhtt commented on org.virtualbox.VirtualBox.metainfo.xml:
```diff
@@ -0,0 +1,93 @@
+    </p>
+    <p>
+      This is an unofficial redistribution of the VirtualBox project, with Cyberus Technology's VirtualBox KVM patches, not including the Extension Packs.
+      Note: VirtualBox is a trademark owned by Oracle. This project has no affiliation with Oracle.
```
Follow the format here https://docs.flathub.org/docs/for-app-authors/requirements#description-notes and should be moved to top.

bbhtt commented on org.virtualbox.VirtualBox.metainfo.xml:
was this submitted upstream?

tulilirockz committed:
feat: update virtualbox flatpak with other repo's code

Signed-off-by: Tulip Blossom <tulilirockz@outlook.com>

tulilirockz committed:
chore(metadata): move unofficial note up

Signed-off-by: Tulip Blossom <tulilirockz@outlook.com>
